pyflakes/checker.py

```diff
@@ -826,7 +848,7 @@ class Checker(object):
     nodeDepth = 0
     offset = None
     traceTree = False
-    _in_annotation = False
+    _in_annotation = None  # other possible values: "bare", "string"
```

Comment:
I would prefer something a little less stringly typed 🤔 -- an enum would be appropriate but we need to support python 2 without dependencies (feel free to ignore -- I can't think of a better way to do this given the constraints)

Reply:
Other ideas I have: Use `""` instead of `None`. Or use numeric constants as enum simulation.

Reply:
oh numeric might be nice, something like:
```python
class AnnotationState:
    NONE = 0
    STRING = 1
    BARE = 2
```

and then when dropping python2 it could be converted to a first-class enum